examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,589 @@
+        kernel_regularizer=None,
+        use_gated=False,
+        **kwargs,
+    ):
+
+        super().__init__(**kwargs)
+
+        self.weight_params_dim = weight_params_dim
+        self.use_gated = use_gated
+
+        self.kernel_initializer = keras.initializers.get(kernel_initializer)
+        self.kernel_regularizer = keras.regularizers.get(kernel_regularizer)
+
+        self.v_init = self.kernel_initializer
+        self.w_init = self.kernel_initializer
+        self.u_init = self.kernel_initializer
+
+        self.v_regularizer = self.kernel_regularizer
+        self.w_regularizer = self.kernel_regularizer
+        self.u_regularizer = self.kernel_regularizer
+
+    def build(self, input_shape):
+
+        # Input Shape
+        # List of 2D tensors with shape: (batch_size, input_dim).
+        input_dim = input_shape[0][1]
+
+        self.v_weight_params = self.add_weight(
+            shape=(input_dim, self.weight_params_dim),
+            initializer=self.v_init,
+            name="v",
+            regularizer=self.v_regularizer,
+            trainable=True,
+        )
+
+        self.w_weight_params = self.add_weight(
+            shape=(self.weight_params_dim, 1),
+            initializer=self.w_init,
+            name="w",
+            regularizer=self.w_regularizer,
+            trainable=True,
+        )
+
+        if self.use_gated:
+            self.u_weight_params = self.add_weight(
+                shape=(input_dim, self.weight_params_dim),
+                initializer=self.u_init,
+                name="u",
+                regularizer=self.u_regularizer,
+                trainable=True,
+            )
+        else:
+            self.u_weight_params = None
+
+        self.input_built = True
+
+    def call(self, inputs):
+
+        # Assigning variables from the number of inputs.
+        instances = [self.compute_weights(instance) for instance in inputs]
+
+        # such that each row summation is equal to 1.
+        alpha = tf.math.softmax(instances, axis=0)
+
+        return [alpha[i] for i in range(alpha.shape[0])]
+
+    def compute_weights(self, instance):
+
+        # in-case "gated mechanism" used.
+        original_instance = instance
+
+        # tanh(v*h_k^T)
+        instance = tf.math.tanh(tf.tensordot(instance, self.v_weight_params, axes=1))
+
+        # for learning non-linear relations efficiently.
+        if self.use_gated:
+
+            instance = instance * tf.math.sigmoid(
+                tf.tensordot(original_instance, self.u_weight_params, axes=1)
+            )
+
+        # w^T*(tanh(v*h_k^T)) / w^T*(tanh(v*h_k^T)*sigmoid(u*h_k^T))
+        return tf.tensordot(instance, self.w_weight_params, axes=1)
+
+
+"""
+## Visualizer tool
+
+Plot the number of bags (given by `PLOT_SIZE`) with respect to the class.
+
+Moreover, if activated, the class label prediction with its associated instance score
+for each bag (after the model has been trained) can be seen.
+"""
+
+# Function for plotting.
```

Comment:
If commenting a function prefer a docstring (and make this lightly more descriptive).

def plot(data, labels, bag_class, predictions=None, attention_weights=None):
  """Utility for plotting bags and attention weights."""

Reply:
Added a docstring for it.